pkg/logql/engine.go

```diff
@@ -419,15 +419,16 @@ func (q *query) JoinSampleVector(next bool, ts int64, r StepResult, stepEvaluato
 				seriesIndex[hash] = series
 			}
 			series.Floats = append(series.Floats, promql.FPoint{
-				T: ts,
+				//T: ts,
```

Comment:
I'm not sure we can do it like this. What do you think, @cstyan?

Reply:
The current code on `main` overrides he timestamps in each vector.

Reply:
isn't the timestamp from the actual vector samples more accurate than the one got from `stepEvaluator.Next()`?

Can you help me understand why it won't make sense?

Reply:
It does make sense for us. I'm just not sure some other component is relying on this overridden timestamp.